src/main/scala/za/co/absa/abris/avro/read/confluent/SchemaManagerFactory.scala

```diff
@@ -58,6 +59,14 @@ object SchemaManagerFactory extends Logging {
           s"'${classOf[AbrisMockSchemaRegistryClient].getCanonicalName}'")
 
         new AbrisMockSchemaRegistryClient()
+      } else if (configs.contains(AbrisConfig.REGISTRY_CLIENT_CLASS)) {
+        val cl = Class.forName(configs(AbrisConfig.REGISTRY_CLIENT_CLASS))
+        logInfo(msg = s"Configuring new Schema Registry instance of type " +
+          s"'${cl.getCanonicalName}'")
+        val instance = cl.getDeclaredConstructor(Array(classOf[util.Map[String, String]]): _*)
+            .newInstance(configs.asJava)
```

Comment:
I think it's a bit too type-unsafe to assume the custom schema registry client will have a constructor that accepts a `Map[String, String]`. For example, the `CachedSchemaRegistryClient` does not meet that criterion. Also, if I want to implement a custom client, it's not enough to implement the `SchemaRegistryClient` interface, but I also need to look in the code to find out what's the expected constructor signature.
I think I would create an abstract class with that constructor and then every custom implementation has to extend that class.

Reply:
That's a very good point! I added a class and a check that the class is a subtype of it before instantiating it.